tests/RobotFramework/tests/cumulocity/smartrest/smartrest_templates.robot

```diff
@@ -0,0 +1,69 @@
+*** Settings ***
+Resource            ../../../resources/common.resource
+Library             Cumulocity
+Library             ThinEdgeIO
+
+Suite Setup         Custom Setup
+Suite Teardown      Get Logs
+
+
+*** Test Cases ***
+
+Add New SmartREST Template
+    [Documentation]  Test to add a new SmartREST template
+
+    # Check if SmartREST template setting exists
+    Verify Non-Existing SmartREST Templates
+
+    # Add Template and check if it was added
+    Execute Command    sudo tedge config set c8y.smartrest.templates template-1
+    Reconnect to Cumulocity
+    Check SmartREST Templates    present    template-1
+
+    # Add Template and check if existing template + added template co-exist
+    Execute Command    sudo tedge config add c8y.smartrest.templates template-2    
+    Reconnect to Cumulocity
+    Check SmartREST Templates    present    template-1    template-2
+
+    # Add Template using set command and check if setting was overwritten
+    Execute Command    sudo tedge config set c8y.smartrest.templates template-2,template-3
+    Reconnect to Cumulocity
+    Check SmartREST Templates    present    template-2    template-3
+    Check SmartREST Templates    not present    template-1
+
+Remove SmartREST Template
+    [Documentation]  Test to remove a SmartREST template
+    Execute Command    sudo tedge config remove c8y.smartrest.templates template-2
+    Reconnect to Cumulocity
+    Check SmartREST Templates    present    template-3
+    Check SmartREST Templates    not present    template-2
+
+    # Remove all SmartREST templates using unset
+    Execute Command    sudo tedge config unset c8y.smartrest.templates
+    Reconnect to Cumulocity
+    Verify Non-Existing SmartREST Templates
+
+
+*** Keywords ***
+
+Custom Setup
+    ${DEVICE_SN}=    Setup
+    Set Suite Variable    $DEVICE_SN
+
+Verify Non-Existing SmartREST Templates
+    ${output}=    Execute Command    tedge config get c8y.smartrest.templates
+    Should Match Regexp    ${output}    ^\s*(\\[\\]|\bnone\b|null)\s*$
```

Comment:
The thoughts behind this were instead of assuming the output will be [], to ensure the keyword can handle different potential outputs for no templates, making the test more robust, not for now but also in the future. I can change it to the state that I had before ` Should Contain  ${output}  []`

Reply:
`c8y.smartrest.templates` is a list of templates, hence `none` and `null` values are to be rejected as errors.